### DISCUSSION TITLE
Bunch not launching apps that have been closed but not quit.

### DISCUSSION THREAD

**Discussion by evanharmon1** [ttscoff/bunch-docs#181] [Q&A] [answered]
Hi! Thanks for the great app!

Launching apps with Bunch works for me when 1) the app is completely closed and 2) when the app was hidden (cmd-h). But when apps are still open but have had their window closed via cmd-w or clicking on the app's red close button (for app's that don't quit the app in that case) Bunch doesn't launch that app. (Bunch doesn't replicate what would happen if I click on an open app that has been closed but not quit.)

Is there a way for Bunch to handle this? Or does this functionality require applescript or something else?

**Comment by ttscoff** [comment]
That's activating, not launching. The app is already launched, so 
sending a launch command has no reaction. Try adding a caret (^) at the 
end of the app name and see if that does what you want. Let me know if 
it doesn't work.

https://bunchapp.co/docs/bunch-files/apps/#activatingapps

-Brett

**Reply by evanharmon1** [reply]
Thanks - activating, not launching is my problem. And I should've specified that yes I've tried the caret. I have tried again and it is consistently not activating my apps (they are not visible if the apps have been "closed" with cmd-w or clicking the red close button). If they are hidden or if the app is completely closed (cmd-q), Bunch works like I expect. I'm just trying something like this so far:
```
---
title: Work - Browse
---

%Safari^
%Firefox^

%Obsidian^
%TaskPaper^
```

**Comment by ttscoff** [comment]
This is an edge request for me, personally, as I never use apps in this 
way. I either hide or quit. That said, there's a specific behavior in 
macOS that presents the default window when an app is brought to 
foreground by clicking the dock icon which I would have to simulate to 
make this work. I'm adding it to the todo list.

In the meantime, you might have luck using:

* tell app "Safari" to activate

(but I haven't tested that, either)

-Brett

**Comment by evanharmon1** [comment]
Ok, I'll try that and see if it works ok. Thanks for the help and the app!
Evan

**Comment by evanharmon1** [comment, marked as answer]
I found a solution that works perfectly so far:
```
%Obsidian^
$ open /Applications/Obsidian.app
```
I'm using this in a variety of ways now and has been 100% reliable for what I was wanting (which is to make the app visible/launched/active from whatever state it may have been in (quit, minimized, hidden). Bunch working great now, thanks!

**Comment by ttscoff** [comment]
Awesome, thanks for the update!

-Brett
________________________________
